coupon.go

```diff
@@ -9,20 +9,23 @@ type CouponDuration string
 // CouponParams is the set of parameters that can be used when creating a coupon.
 // For more details see https://stripe.com/docs/api#create_coupon.
 type CouponParams struct {
-	Params
-	Duration                                     CouponDuration
-	ID                                           string
-	Currency                                     Currency
-	Amount, Percent, DurationPeriod, Redemptions uint64
-	RedeemBy                                     int64
+	Params         `form:"*"`
+	Duration       CouponDuration `form:"duration"`
+	ID             string         `form:"id"`
+	Currency       Currency       `form:"currency"`
+	Amount         uint64         `form:"amount_off"`
+	Percent        uint64         `form:"percent_off"`
+	DurationPeriod uint64         `form:"duration_in_months"`
+	Redemptions    uint64         `form:"max_redemptions"`
+	RedeemBy       int64          `form:"redeem_by"`
 }
 
 // CouponListParams is the set of parameters that can be used when listing coupons.
 // For more detail see https://stripe.com/docs/api#list_coupons.
 type CouponListParams struct {
-	ListParams
-	Created      int64
-	CreatedRange *RangeQueryParams
+	ListParams   `form:"*"`
+	Created      int64             `form:"created"`
+	CreatedRange *RangeQueryParams `form:"created"`
```

Comment:
Hm, I'm honestly not sure whether `Created` is 100% supported, but you're right in that it's very broad. Once again though, can we leave that change for another PR?

Reply:
And another note on that one: hoisting `Created` will mean that it gets accessed like `params.ListParams.Created = ...` instead of the current `params.Created = ...` which seems unnecessarily impactful for now.

Reply:
I'm definitely not asking for the change now. But this PR touches almost every single file in the library. I'm not sure if we will make such an impactful change again in the next 6 months so it was the best solution for me to raise things I had never realized before because it's the only way to flag so that we can schedule improvements soon.
Sorry for the confusion.